lxd/db/query/slices.go

```diff
@@ -4,8 +4,40 @@ import (
 	"database/sql"
 	"fmt"
 	"strings"
+
+	"github.com/lxc/lxd/shared/version"
+	"github.com/pkg/errors"
 )
 
+// SelectURIs returns a list of LXD API URI strings for the resource with the
+// given name.
+func SelectURIs(stmt *sql.Stmt, resource string) ([]string, error) {
+	rows, err := stmt.Query()
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to query %s URIs", resource)
+	}
+	defer rows.Close()
+
+	uris := []string{}
+
+	for rows.Next() {
+		var name string
+		err := rows.Scan(&name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to scan %s URI", resource)
+		}
+		uri := fmt.Sprintf("/%s/%s/%s", version.APIVersion, resource, name)
```

Comment:
Won't work for snapshots, backups and storage-volumes, probably needs some hardcoded special casing for those.